### ISSUE TITLE
Make systemd restart the server when i install new mod

### ISSUE THREAD

**Issue by Emojigit** [minetest/minetest#10264]
> > > If i can [run tow (or more) server on the same device](https://github.com/minetest/minetest/issues/10204), can i run two versions on the same device?
> > > The two version that i want:
> > > 
> > > * 5.3.0 (Download from ppa:minetestdevs/stable)
> > > * 5.4.0-dev (build from source code)
> > > 
> > > RAM: 4GB
> > > Ubuntu Version: 20.04
> > 
> > 
> > I believe you can, just have a process nameA and process nameB. Such as mainserv and devserver for example.
> > On Linux this means making a server1.service and a server2.service file. Then you can simply do systemctl enable server1
> > systemctl enable server2
> > etc, etc etc
> > Oh yea don't forget to assign DIFFERENT ports OR IPs.
> 
> I never use systemd to start the server, I always use command to start it.
> How to set up a server that use systemd to start?

So firstly you will want to with your text editor (like vim, nano, etc) under your ROOT user create a file for each of your server and input the following.... 
Server1:/etc/systemd/system/server1.service  
```
[Unit]
Description=Minetest Server1
After=network.target

[Service]
Type=simple
User=youruser
Group=yourgroup
WorkingDirectory=/home/minetest
ExecStart=/usr/bin/minetest --server
Restart=on-failure

RestartSec=5s
[Install]
WantedBy=multi-user.target
```
Server2:/etc/systemd/system/server2.service 
```
[Unit]
Description=Minetest Server2
After=network.target

[Service]
Type=simple
User=youruser
Group=yourgroup
WorkingDirectory=/path/to/minetestserver2
ExecStart=/path/to/server2/executable/minetest --server
Restart=on-failure

RestartSec=5s
[Install]
WantedBy=multi-user.target
```
_Just as long you configure the second one properly it should allow you to run both versions properly_. NOW to make these systemd files executable (and works on boot)... 
systemctl enable minetestserver1.service
systemctl enable minetestserver2.service 

To start/stop/restart them just replace the "enable" on the server(s) you wanna start/stop/restart on.  

Hope this helps you out!

_Originally posted by @FriendlyGamer in https://github.com/minetest/minetest/issues/10205#issuecomment-660563213_
<hr>
My server is keep adding mod. How to make these systemctl script restart the server when i install and enable new mod?

**Comment by SmallJoker** [comment]
TL;DR version: Read the last sentence of the post above.


I dislike this idea - AFAIK there's no other service that restarts itself after changing the configuration. A Minetest service should only be interrupted (stop/restart) manually. This avoids accidental restarts when the file is `touch`ed, or nothing relevant was added to the file.
:-1:

**Comment by rubenwardy** [comment]
I have a script to restart the server when a mod update is detected. It uses git to update the mods. You should do this, rather than detecting file system changes

**Comment by tenplus1** [comment]
It's probably a lot easier to have a server .sh script that runs your minetest server and on stop/failure backs up logs and does a restart loop...  This way you can issue a shutdown inside the game itself to restart as well.

**Comment by paramat** [comment]
Seems a question unrelated to MT development, as i see no feature request, should probably have been asked in the forum.
You are welcome to continue discussion, but closing.
